File: src/2535.difference-between-element-sum-and-digit-sum-of-an-array.2624/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var differenceOfSum = function (nums) {
    const elementSum = nums.reduce((sum, item) => sum + item, 0);
    const digitSum = nums.reduce((sum, item) => {
        while (item) {
            const digit = item % 10;
            sum += digit;
            item = (item - digit) / 10;
        }

        return sum;
    }, 0);

    return Math.abs(elementSum - digitSum);
};

File: src/2535.difference-between-element-sum-and-digit-sum-of-an-array.2624/solution.java
class Solution {
    public int differenceOfSum(int[] nums) {
        int result = 0;
        for (int n : nums) {
            result += n;
            while (n > 0) {
                result -= n % 10;
                n /= 10;
            }
        }

        return Math.abs(result);
    }
}